ttnn/cpp/ttnn/operations/data_movement/repeat/device/repeat_device_operation.cpp

```diff
@@ -27,43 +39,63 @@ void RepeatDeviceOperation::validate(const std::vector<Tensor>& input_tensors) c
+        operation_attributes.m_output_mem_config.memory_layout() == input_tensor_a.memory_config().memory_layout(),
         "Output tensor must have the same memory layout as input tensor");
 }
 
-std::vector<TensorSpec> RepeatDeviceOperation::compute_output_specs(const std::vector<Tensor>& input_tensors) const {
-    const auto& input_tensor_a = input_tensors.at(0);
+void RepeatDeviceOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_on_program_cache_miss(operation_attributes, tensor_args);
+}
+
+RepeatDeviceOperation::spec_return_value_t RepeatDeviceOperation::compute_output_specs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& input_tensors) {
+    const auto& input_tensor_a = input_tensors.input;
     auto output_shape = input_tensor_a.logical_shape();
-    output_shape[m_is_last_dim ? -1 : 1] *= m_num_repeats;
+    output_shape[operation_attributes.m_is_last_dim ? -1 : 1] *= operation_attributes.m_num_repeats;
 
-    auto mem_config = this->m_output_mem_config;
+    auto mem_config = operation_attributes.m_output_mem_config;
     if (input_tensor_a.memory_config().is_sharded()) {
         auto shard_spec = input_tensor_a.shard_spec().value();
         shard_spec.shape[0] = output_shape[0];
         mem_config = mem_config.with_shard_spec(shard_spec);
     }
-    return {TensorSpec(
+    return TensorSpec(
         output_shape,
         tt::tt_metal::TensorLayout(
-            input_tensor_a.dtype(), tt::tt_metal::PageConfig(input_tensor_a.layout()), mem_config))};
+            input_tensor_a.dtype(), tt::tt_metal::PageConfig(input_tensor_a.layout()), mem_config));
+}
+
+RepeatDeviceOperation::tensor_return_value_t RepeatDeviceOperation::create_output_tensors(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& input_tensors) {
+    return create_device_tensor(
+        compute_output_specs(operation_attributes, input_tensors), input_tensors.input.device());
 }
 
 tt::tt_metal::operation::OpPerformanceModelGeneral<std::vector<Tensor>>
 RepeatDeviceOperation::create_op_performance_model(
-    const std::vector<Tensor>& input_tensors,
-    const std::vector<std::optional<const Tensor>>& optional_input_tensors,
-    std::vector<Tensor>& output_tensors) const {
-    const auto& input_tensor = input_tensors.at(0);
-    const auto& output_tensor = output_tensors.at(0);
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output_tensor) const {
+    const auto& input_tensor = tensor_args.input;
     int ideal_dev_clock_cycles = operations::data_movement::common_tm_bw_model(input_tensor, output_tensor);
     tt::tt_metal::operation::OpPerformanceModelGeneral<std::vector<Tensor>> result(
-        input_tensors, output_tensors, ideal_dev_clock_cycles);
+        {input_tensor}, {output_tensor}, ideal_dev_clock_cycles);
```

Comment:
After changing the template argument to `tensor_return_value_t` this is probably how the constructor argument list should look like. @philei-tt please take a look.
```suggestion
        {input_tensor}, output_tensor, ideal_dev_clock_cycles);
```

Here's the definition of the `OpPerformanceModelGeneral ` constructor.
https://github.com/tenstorrent/tt-metal/blob/39372492b282c696a7022a9c24a28ba3e81cde4c/ttnn/api/ttnn/operation.hpp#L139